torch/csrc/jit/tensorexpr/ir_printer.cpp

```diff
@@ -191,25 +191,26 @@ void IRPrinter::visit(const CompareSelectPtr& v) {
   withParens(v->ret_val2());
 }
 
-static void formatFPSuffix(std::ostream& os, double v) {
-  os << (v == std::ceil(v) ? ".0" : "");
+static void formatFPSuffix(std::ostream& os, double v, bool flag) {
+  os << (flag && v == std::ceil(v) ? ".0" : "");
 }
 
 template <typename T>
-static void formatFPSuffix(std::ostream& os, T v) {
-  os << (v == std::ceil(v) ? ".f" : "f");
+static void formatFPSuffix(std::ostream& os, T v, bool flag) {
+  os << (flag && v == std::ceil(v) ? ".f" : "f");
 }
 
 template <typename T, std::enable_if_t<std::is_floating_point_v<T>>* = nullptr>
 static void formatImm(std::ostream& os, T v) {
   const int precision = 16;
+  const T const_range[] = {static_cast<T>(-1e16), static_cast<T>(1e16)};
```

Comment:
Do you think it is better to reuse precision?

```c++
  constexpr int precision = 16;
  constexpr T lower_bound = static_cast<T>(-std::pow(10, -precision));
  constexpr T upper_bound = -lower_bound ;
  const T const_range[] = {lower_bound, upper_bound};
```

Reply:
It is better, but the second line,  `-std::pow(10, -precision)` should be `-std::pow(10, precision)`.